llvm/lib/Target/BPF/BPFISelLowering.cpp

```diff
@@ -726,6 +733,38 @@ SDValue BPFTargetLowering::LowerATOMIC_LOAD_STORE(SDValue Op,
   return Op;
 }
 
+SDValue BPFTargetLowering::LowerTRAP(SDValue Op, SelectionDAG &DAG) const {
+  MachineFunction &MF = DAG.getMachineFunction();
+  Function &F = MF.getFunction();
+  if (F.hasFnAttribute(Attribute::Naked))
+    return Op;
```

Comment:
Probably this should be part of the generic handling for TrapUnreachable. IMHO we should never insert extra code into a naked function, for other targets as well.

Reply:
Sounds good to me. Let me check more and may have a patch for this later.